packages/teams-js/src/public/location.ts

```diff
@@ -46,82 +51,147 @@ export namespace location {
   }
 
   /**
-   * Fetches current user coordinates or allows user to choose location on map
-   *
-   * @param props {@link LocationProps} - Specifying how the location request is handled
-   * @returns Promise that will be fulfilled when the operation has completed
+   * Fetches current user coordinates
+   * @returns User's current location
    */
-  export function getLocation(props: LocationProps): Promise<Location>;
+  export function getCurrentLocation(): Promise<Location> {
+    ensureInitialized(FrameContexts.content, FrameContexts.task);
+    if (!isSupported()) {
+      throw errorNotSupportedOnPlatform;
+    }
+    return sendAndHandleError('location.getLocation', { allowChooseLocation: false, showMap: false });
+  }
+
   /**
-   * @deprecated
-   * As of 2.0.0, please use {@link location.getLocation location.getLocation(props: LocationProps): Promise\<Location\>} instead.
-   * @param props {@link LocationProps} - Specifying how the location request is handled
-   * @param callback - Callback to invoke when current user location is fetched
+   * Checks whether or not location has user permission
+   *
+   * @returns if the location has user permission
    */
-  export function getLocation(props: LocationProps, callback: (error: SdkError, location: Location) => void): void;
-  export function getLocation(
-    props: LocationProps,
-    callback?: (error: SdkError, location: Location) => void,
-  ): Promise<Location> {
+  export function hasPermission(): Promise<boolean> {
     ensureInitialized(FrameContexts.content, FrameContexts.task);
+    if (!isSupported()) {
+      throw errorNotSupportedOnPlatform;
+    }
+    const permissions: DevicePermission = DevicePermission.GeoLocation;
 
-    return callCallbackWithErrorOrResultFromPromiseAndReturnPromise<Location>(getLocationHelper, callback, props);
+    return new Promise<boolean>(resolve => {
+      resolve(sendAndHandleError('permissions.has', permissions));
+    });
   }
 
-  function getLocationHelper(props: LocationProps): Promise<Location> {
-    return new Promise<Location>(resolve => {
-      if (!isCurrentSDKVersionAtLeast(locationAPIsRequiredVersion)) {
-        throw { errorCode: ErrorCode.OLD_PLATFORM };
-      }
-      if (!props) {
-        throw { errorCode: ErrorCode.INVALID_ARGUMENTS };
-      }
-      if (!isSupported()) {
-        throw errorNotSupportedOnPlatform;
-      }
-      resolve(sendAndHandleError('location.getLocation', props));
+  /**
+   * Request user permission for location
+   *
+   * @returns if the user conseted permission for location
+   */
+  export function requestPermission(): Promise<boolean> {
+    ensureInitialized(FrameContexts.content, FrameContexts.task);
```

Comment:
Do you think it would make sense to put the implementations of hasPermission and requestPermission into a shared helper function so we don't have to repeat the logic in every capability that uses them?

Reply:
Like we talked, there is not much to share in these functions, it is good to keep them in separate files :)